### PR TITLE
[Azure Spring Apps] Add StandardGen2 tier

### PR DESCRIPTION
**Pull request by ShichaoQiu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by ShichaoQiu** [comment]
Hi @zhoxing-ms , would you pls help to review this PR? We plan to release this PR on 3/15, pls **don't merge** this PR before that.

File: src/spring/azext_spring/app.py [hunk 1]
```diff
@@ -78,6 +87,17 @@ def app_create(cmd, client, resource_group, service, name,
     logger.warning(LOG_RUNNING_PROMPT)
     _ensure_app_not_exist(client, resource_group, service, name)
     sku = get_spring_sku(client, resource_group, service)
+
+    if sku.tier.upper() == 'STANDARDGEN2':
+        if cpu is None and memory is None:
+            cpu = '500m'
+            memory = '1Gi'
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/app.py:94]: Why isn't this logic like this?
```
       if cpu is None:
            cpu = '500m'
       if memory is None:
            memory = '1Gi'
```

File: src/spring/azext_spring/app.py [hunk 2]
```diff
@@ -560,5 +607,10 @@ def _fulfill_deployment_creation_options(skip_clone_settings, client, resource_g
             options.update(deployment_source_options_from_resource(active_deployment))
     if not options.get('sku', None):
         options['sku'] = get_spring_sku(client, resource_group, service)
+
+    if options['sku'].tier.upper() == 'STANDARDGEN2' and skip_clone_settings and kwargs['cpu'] is None and kwargs['memory'] is None:
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/app.py:611]: Could you please split this complex judgment statement into multiple variables with better readability?

File: src/spring/azext_spring/_marketplace.py [hunk 3]
```diff
@@ -14,8 +14,8 @@
 def _spring_list_marketplace_plan(cmd, client):
     # return get_mgmt_service_client(cli_ctx, AppPlatformManagementClient_20220501preview)
     from azure.cli.core.commands.client_factory import get_mgmt_service_client
-    from .vendored_sdks.marketplace.v2022_11_01_preview import MarketplaceRPService
-    from .vendored_sdks.marketplace.v2022_11_01_preview.models import Offer
+    from .vendored_sdks.marketplace.v2023_01_01_preview import MarketplaceRPService
+    from .vendored_sdks.marketplace.v2023_01_01_preview.models import Offer
```
Comment (zhoxing-ms) [inline src/spring/azext_spring/_marketplace.py:18]: I have a small suggestion: if the SDK path we import does not contain the api-version part, then when upgrading the SDK version in the future, we only need to replace the SDK folder, not all the import paths.

File: src/spring/azext_spring/app.py (hunk 1 above)
Comment (ShichaoQiu) [reply src/spring/azext_spring/app.py:94]: Good question, actually, the background is that there is a hard limit for new tier - StandardGen2 that the value of Memory must be double value of CPU, it means that, for example, if the cpu is 500m (0.5 cpu), the memory must be 0.5 * 2 = 1 Gi. 

The logic here is that if customer don't input any value for cpu and memory, we will use the default value (500m, 1Gi)，but if they input any value of CPU or Memory, we won't do some special logic to handle the other one, which will be handled by our backend service.

File: src/spring/azext_spring/app.py (hunk 2 above)
Comment (ShichaoQiu) [reply src/spring/azext_spring/app.py:612]: I have added a comment to explain the logic here

File: src/spring/azext_spring/_marketplace.py (hunk 3 above)
Comment (ShichaoQiu) [reply src/spring/azext_spring/_marketplace.py:18]: Very very good idea. As you can see, there are multi api version in our CLI extension, we plan to only use the latest API version in the future. There are 2 steps here.
1. Make sure we only use 1 API version in the code
2. Remove unused the API versions and remove the version info from the SDK file path.

We will create another PR to do the step 2.

Comment (zhoxing-ms) [reply src/spring/azext_spring/_marketplace.py:18]: OK, sounds good!

File: src/spring/azext_spring/app.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/spring/azext_spring/app.py:94]: Got it, thanks~

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**